src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -854,11 +859,14 @@ def apply_inputs_mapping(
                 result[map_to_key] = map_value  # Literal value
         # Return all not found mapping relations in one exception to provide better debug experience.
         if notfound_mapping_relations:
-            raise MappingSourceNotFound(
-                f"Couldn't find these mapping relations: {', '.join(notfound_mapping_relations)}. "
+            invalid_relations = ", ".join(notfound_mapping_relations)
+            raise InputMappingError(
+                message_format="The input for flow is incorrect. "
+                "Couldn't find these mapping relations: {invalid_relations}. "
                 "Please make sure your input mapping keys and values match your YAML input section and input data. "
-                "If a mapping value has a '${data' prefix, it might be generated from the YAML input section, "
-                "and you may need to manually assign input mapping based on your input data."
+                "If a mapping reads input from 'data', it might be generated from the YAML input section, "
+                "and you may need to manually assign input mapping based on your input data.",
```

Comment:
Let's add a `()` to multi-line messages.
Like this:
![image](https://github.com/microsoft/promptflow/assets/113209/00a1cfef-eadf-4204-8353-3a97ccfdd959)

Reply:
Fixed.